src/components/MovieDetails.jsx

```diff
@@ -1,35 +1,57 @@
 import PropTypes from 'prop-types';
 import CONSTANTS from '../constants/constants';
+import { Link, withRouter } from "react-router-dom";
+import { connect } from 'react-redux';
+import { getMovie } from '../actions/actionCreator';
+import { baseURL } from '../constants/baseURL';
 
-const MovieDetails = ({ movie }) => {
-  const MovieDetails = movie ? (
-    <div className="movie-wrapper">
-      <div className="movie-header">
-        <h5 className="red-text text-darken-2">{CONSTANTS.NETFLIX}</h5>
-        <button className="red lighten-1 btn right">{CONSTANTS.SEARCH}</button>
-      </div>
-      <div className="movie-poster">
-        <img className="thumbnails-details" src={movie.poster_path} />
-      </div>
-      <div className="movie-detail">
-        <h3 className="red-text text-darken-2 title-details">{movie.title}</h3>
-        <h5 className="white-text text-darken-2 avg-rating">{movie.vote_average}</h5>
-        <p className="white-text text-darken-2">{movie.genres.join(" & ")}</p>
-        <span className="white-text text-darken-2">{movie.release_date}</span>
-        <span className="white-text text-darken-2">{movie.runtime + "min"}</span>
-        <p className="white-text text-darken-2">{movie.overview}</p>
+class MovieDetails extends Component {
+
+  componentDidMount() {
+    let url = `${baseURL}/${this.props.match.params.id}`;
+    this.props.getMovie(url);
+  }
+
+  render() {
+    const movie = this.props.selectedMovie ? this.props.selectedMovie[0] : null;
+    const MovieDetails = movie ? (
+      <div className="movie-wrapper">
+        <div className="movie-header">
+          <h5 className="red-text text-darken-2">{CONSTANTS.NETFLIX}</h5>
+          <Link to={'/'}>
+            <button className="red lighten-1 btn right">{CONSTANTS.SEARCH}</button>
+          </Link>
+        </div>
+        <div className="movie-poster">
+          <img className="thumbnails-details" src={movie.poster_path} />
+        </div>
+        <div className="movie-detail">
+          <h3 className="red-text text-darken-2 title-details">{movie.title}</h3>
+          <h5 className="white-text text-darken-2 avg-rating">{movie.vote_average}</h5>
+          <p className="white-text text-darken-2">{movie.genres.join(" & ")}</p>
+          <span className="white-text text-darken-2">{movie.release_date}</span>
+          <span className="white-text text-darken-2">{movie.runtime + "min"}</span>
+          <p className="white-text text-darken-2 overview">{movie.overview}</p>
+        </div>
       </div>
-    </div>
-  ) : null;
+    ) : null;
 
-  return (
-    <div>{MovieDetails}</div>
-  )
+    return (
+      <div>{MovieDetails}</div>
+    )
+  }
 }
 
 MovieDetails.propTypes = {
   movie: PropTypes.object
 };
 
-export default MovieDetails;
+const mapDispatchToProps = { getMovie };
+
+const mapStateToProps = (state, ownProps) => ({
+  movie: state.movies.data.find(({ id }) => id === Number(ownProps.match.params.id)),
```

Comment:
I think `movie` prop is not used anymore and can be removed.

Reply:
Removed.